### ISSUE TITLE
npm install hangs after installing all dependencies from package.json Node Js    

### ISSUE THREAD

**Issue by viveksharmaui** [docker/toolbox#833]
Hi ,
I'm using docker for windows 7 i have created dockerFile which is simple file for installing dependencies etc . 
When build occur's on npm install step , i install all dependencies correctly but when installation is completed build does not move to next step of dockerFile.

**Comment by RakeshVenkat** [comment]
@viveksharmaui If I am allowed to comment on your issue, I just wanted to ask did you get this issue to be resolved? Even I face the same issue on a Windows 2016 server (base image: stefanscherer/node-windows)

**Comment by viveksharmaui** [comment]
I didn't find any solution i think the problem is in my system and not in image .

**Comment by gwartnes** [comment]
I'm also having this same exact issue...been working through it for the past few hours with no success.

**Comment by viveksharmaui** [comment]
Wow that great ! did you find the reason why it's happen ?

**Comment by gwartnes** [comment]
> Wow that great ! did you find the reason why it's happen ?

Nope. But after a subsequent try, it took much less time to continue, maybe about 1 or 2 minutes.

**Comment by viveksharmaui** [comment]
Great !

**Comment by lagneshthakur** [comment]
It did not work for me even after waiting more than 20 minutes.

**Comment by viveksharmaui** [comment]
Simply uninstall docker toolbox i think members are not maintaining it .

**Comment by lagneshthakur** [comment]
Turns out the VPN was the issue. So if anyone faces this over VPN, should switch that off and try again.

I got it working without the VPN, thanks @viveksharmaui

**Comment by rong0312** [comment]
Having the same issue with windows 10. 'npm install' hangs the whole build. 
Got to be node_modules issue between layers switch

**Comment by vladimirze** [comment]
Not using docker toolbox, but I found this issue by looking for solution to the same problem.

Also I found this: [Docker build (Windows) hangs after RUN 'npm install'](https://stackoverflow.com/questions/58936278/docker-build-windows-hangs-after-run-npm-install)

Since I am working on frontend project, after the build completes I can remove `node_modules/`:
```
RUN npm install && npm run build && rm node_modules/
```
This resolved the 15min wait that I had between this RUN and the next one.

**Comment by viveksharmaui** [comment]
> Not using docker toolbox, but I found this issue by looking for solution to the same problem.
> 
> Also I found this: [Docker build (Windows) hangs after RUN 'npm install'](https://stackoverflow.com/questions/58936278/docker-build-windows-hangs-after-run-npm-install)
> 
> Since I am working on frontend project, after the build completes I can remove `node_modules/`:
> 
> ```
> RUN npm install && npm run build && rm node_modules/
> ```
> 
> This resolved the 15min wait that I had between this RUN and the next one.

@vladimirze according to the command you are running build it is suitable for production but what if I'm working in development ? i need to create a build each time ?

**Comment by vladimirze** [comment]
> @vladimirze according to the command you are running build it is suitable for production but what if I'm working in development ? i need to create a build each time ?

Yes, it was for production. 
Not sure have to solve that but maybe you can use cache (docker run without `--no-cache`) and rebuild it only when one of you dependencies was changed.
